src/Native/gen-buildsys-win.bat

```diff
@@ -27,14 +24,16 @@ for /f "delims=" %%a in ('powershell -NoProfile -ExecutionPolicy ByPass "& .\pro
 popd
 
 :DoGen
-"%CMakePath%" "-DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%" "-DCMAKE_INSTALL_PREFIX=%__CMakeBinDir%" "-DMKL_LIB_PATH=%MKL_LIB_PATH%" -G "Visual Studio %__VSString%" -B. -H%1
+
+if /i "%3" == "x86"     ("%CMakePath%" "-DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%" "-DCMAKE_INSTALL_PREFIX=%__CMakeBinDir%" "-DMKL_LIB_PATH=%MKL_LIB_PATH%" -G "Visual Studio %__VSString%" -B. -H%1)
+if /i "%3" == "x64"     ("%CMakePath%" "-DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%" "-DCMAKE_INSTALL_PREFIX=%__CMakeBinDir%" "-DMKL_LIB_PATH=%MKL_LIB_PATH%" -G "Visual Studio %__VSString%" -A "x64" -B. -H%1)
```

Comment:
instead of duplicating the whole line, you can define an additional variable something like "arch" and set it to "-A "x64" in if /i "%3" == "x64" and for the other case it can be empty then you can plug that in in this command line and not having to duplicate it. See here for what I'm talking about [https://github.com/dotnet/corefx/blob/6f9570e012054bf407778c0b07632e7733ecbb13/src/Native/Windows/gen-buildsys-win.bat#L23](https://github.com/dotnet/corefx/blob/6f9570e012054bf407778c0b07632e7733ecbb13/src/Native/Windows/gen-buildsys-win.bat#L23)